docs/IgorFromUA/scripts/games.js

```diff
@@ -0,0 +1,152 @@
+    const fragment = new DocumentFragment();
+    arrGames.forEach((game) => {
+        fragment.append(createCardElement(game));
+    });
+    container.append(fragment);
+}
+
+function loadingSwitch(switching, place) {
+    const LOADING_ELEMENT = document.querySelector('[data-loading]');
+    if (switching) place.textContent = '';
+    LOADING_ELEMENT.style.visibility = switching ? 'visible' : 'hidden';
+}
+
+function getGames(url, place) {
+    loadingSwitch(true, place);
+    const optionsGamesUrl = {
+        method: 'GET',
+        headers: {
+            'X-RapidAPI-Key': '1c3169c707mshb51bff34cbc9ff6p1749b9jsn648a19134256',
+            'X-RapidAPI-Host': 'mmo-games.p.rapidapi.com',
+        },
+    };
+    return fetch(url, optionsGamesUrl)
+        .then((response) => {
+            if (response.status < 200 || response.status >= 300) {
+                throw new Error(response.status);
+            } else {
+                return response.json();
+            }
+        })
+        .then((result) => result.slice(0, 50))
+        .then((result) => result.map((game) => {
+            const copyGame = { ...game };
+            copyGame.small_description = `${game.short_description.slice(0, 52)}...`;
+            return copyGame;
+        }))
+        .catch((error) => {
+            place.insertAdjacentHTML('afterbegin', `<span class="error-message">Something went wrong! ERROR: ${error.message}</span>`);
+            console.log('get error', error);
+        })
+        .finally(() => loadingSwitch(false));
+}
+
+async function applyFilter(url, place) {
+    try {
+        const urlGamesQuery = `${url}?${creatSearchQueryString()}`;
+        const isOldGame = document.querySelector('[data-age-games="old"]').checked;
+        const games = await getGames(urlGamesQuery, place);
+        if (isOldGame) games.reverse();
+        const searchTerm = document.querySelector('[data-filter-search]').value;
+        const filterArrGames = filterGames(games);
+        const markedTextFilterArrGames = filterArrGames.map((game) => {
+            const copyGame = { ...game };
+            copyGame.title = markedSearchText(searchTerm, game.title);
+            copyGame.small_description = markedSearchText(searchTerm, game.small_description);
+            return copyGame;
+        });
+        renderCards(place, markedTextFilterArrGames);
+    } catch (e) {
+        console.log('filter error: ', e);
+    }
+}
+
+async function init() {
+    const urlGames = 'https://mmo-games.p.rapidapi.com/games';
+    const CARDS_LIST = document.querySelector('[data-cards]');
+    try {
+        const games = await getGames(urlGames, CARDS_LIST);
+        if (!Array.isArray(games)) throw new Error('is not Array');
+        renderCards(CARDS_LIST, games);
+    } catch (e) {
+        console.log('init error: ', e.message);
+    } finally {
+        const [INPUT_GENGE, INPUT_PLATFORM, RADIO_NEW, RADIO_OLD, , BUTTON_APPLY] = Array.from(document.forms['filter-form']);
```

Comment:
So, if you change the order of form elements in HTML, your JS logic will be broken.
The better approach is to fix, which element you need by `data-attribute`

Reply:
Also would be good to move lines 143-148 to some function, let's say `attachHandlers`